src/components/languageSelect/index.js

```diff
@@ -5,15 +5,19 @@ import Menu from "../menu"
 
 import "./style.css"
 
-const {REACT_APP_DEFAULT_LANGUAGE} = process.env
-
 const LanguageSelect = () => {
-  const [selectedLanguage, setSelectedLanguage] = useState(REACT_APP_DEFAULT_LANGUAGE)
+  const [selectedLanguage, setSelectedLanguage] = useState(i18n.language)
   const [isMenuVisible, setIsMenuVisible] = useState(false)
 
+  const fallbackLng = i18n.options.fallbackLng[0]
+
   useEffect(() => {
-    i18n.changeLanguage(selectedLanguage)
-  }, [selectedLanguage])
+    if (i18n.availableLanguages.includes(selectedLanguage)) {
+      i18n.changeLanguage(selectedLanguage)
+    } else {
+      i18n.changeLanguage(fallbackLng)
+    }
```

Comment:
Maybe I would put this if else into its own function, for the rest looks good to me.

Reply:
Are you thinking something along these lines? That particular logic isn't going to be reused and I'm not convinced this is any clearer in what it's doing.
```jsx
  useEffect(() => {
    i18n.changeLanguage(returnValidLanguage(selectedLanguage))
  }, [selectedLanguage, fallbackLng])

  ...

  const returnValidLanguage = (selectedLanguage) => {
  const fallbackLng = i18n.options.fallbackLng[0]

  if (i18n.availableLanguages.includes(selectedLanguage)) {
   return selectedLanguage
  } 
  return fallbackLng
}
```